python/cudf/cudf/core/dataframe.py

```diff
@@ -6241,13 +6229,32 @@ def _reduce(
                     dtype="float64",
                 )
         if axis in {0, 2}:
-            if axis == 2 and op in ("kurtosis", "kurt", "skew"):
-                # TODO: concat + op can probably be done in the general case
-                # for axis == 2.
-                # https://github.com/rapidsai/cudf/issues/14930
+            if axis == 2 and op in (
+                "kurtosis",
+                "kurt",
+                "skew",
+                "mean",
+                "sum",
+                "product",
+            ):
+                return getattr(concat_columns(source._data.columns), op)(
+                    **kwargs
+                )
+            elif axis == 2 and op in {"std", "var"}:
```

Comment:
How should std and var reductions be treated? I think 1. is right, but I wanted to check.
1. Combine the columns, then compute the reduction
2. Compute the reduction over columns, then compute the reduction again over those results
cc. @mroeschke, @vyasr

Reply:
In regards to 1, we do have to keep this comment in mind: https://github.com/rapidsai/cudf/issues/14930#issuecomment-1935124143

In terms of correctness, I forgot if std and var validate the minimum number of elements in the columns, but if so, the logic for `axis=None` should be something like.

1. `if self.size < minimum_std_var_elements`: Return NA
1. `if len(self) < minimum_std_var_elements and self.size > minimum_std_var_elements`: Your option 1
2. `else`: Your option 2